File: spikes/splfantasy/src/main/kotlin/com/nicolaspetras/splfantasy/model/scoring/SplMatchScore.kt
package com.nicolaspetras.splfantasy.model.scoring

import com.nicolaspetras.splfantasy.model.SplTeamName

data class SplMatchScore(
    val homeTeamName: SplTeamName,
    val awayTeamName: SplTeamName,
    val homeTeamScore: Int,
    val awayTeamScore: Int,
    val homeTeamScores: ArrayList<SplPlayerMatchScore>,
    val awayTeamScores: ArrayList<SplPlayerMatchScore>
)

File: frontend/src/app/model/SplMatchScore.ts
export class SplMatchScore{
  homeTeamName: string;
  awayTeamName: string;
  homeTeamScore: number;
  awayTeamScore: number;
  homeTeamScores: Array<SplMatchScore>;
  awayTeamScores: Array<SplMatchScore>


  constructor(homeTeamName: string = "None", awayTeamName: string = "None", homeTeamScore: number = 0,
              awayTeamScore: number = 0, homeTeamScores: Array<SplMatchScore> = [],
              awayTeamScores: Array<SplMatchScore> = []) {
    this.homeTeamName = homeTeamName;
    this.awayTeamName = awayTeamName;
    this.homeTeamScore = homeTeamScore;
    this.awayTeamScore = awayTeamScore;
    this.homeTeamScores = homeTeamScores;
    this.awayTeamScores = awayTeamScores;
  }
}
